pkg/skaffold/watch/git_repo.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const masterRef = "refs/heads/master"
+
+type GitFileWatcher struct {
+	url          string
+	pollInterval time.Duration
+
+	// TODO(r2d4): watch pulls and tags
+}
+
+func NewGitFileWatcher(url string, pollInterval time.Duration) FileWatcher {
+	logrus.Infof("Using git file watcher: %s", url)
+	return &GitFileWatcher{
+		url:          url,
+		pollInterval: pollInterval,
+	}
+}
+
+func (g *GitFileWatcher) Run(ctx context.Context, callback FileChangedFn) error {
```

Comment:
All the code here considers git is installed on the machine. We try to avoid that requirement in the git tagger. I'm ok with that and I'm wondering if we shouldn't maybe get rid of go-git because it's not a high fidelity clone of git anyways.

Reply:
Yeah I initially tried to use it but it didnt have `ls-remote` or `diff` support.